src/types/chrono.rs

```diff
@@ -0,0 +1,374 @@
+        let date_str = self.format("%Y-%m-%d %H:%M:%S%.f%:z").to_string();
+        date_str.bind_parameter(stmt, col)
+    }
+}
+
+/// "YYYY-MM-DD HH:MM:SS.SSS[+-]HH:MM"/"YYYY-MM-DD HH:MM"/"YYYY-MM-DD HH:MM:SS"/"YYYY-MM-DD HH:MM:SS.SSS"/ Julian Day / Unix Time => ISO 8601 date and time with time zone.
+/// ("YYYY-MM-DDTHH:MM:SS.SSS[+-]HH:MM"/"YYYY-MM-DDTHH:MM"/"YYYY-MM-DDTHH:MM:SS"/"YYYY-MM-DDTHH:MM:SS.SSS" also supported)
+/// When the timezone is not specified, UTC is used.
+impl FromSql for DateTime<UTC> {
+    unsafe fn column_result(stmt: *mut sqlite3_stmt, col: c_int) -> Result<DateTime<UTC>> {
+        match sqlite3_column_type(stmt, col) {
+            ffi::SQLITE_TEXT => {
+                let s = try!(String::column_result(stmt, col));
+                if s.len() > 23 {
+                    let fmt = if s.as_bytes()[10] == b'T' {
+                        "%Y-%m-%dT%H:%M:%S%.f%:z"
+                    } else {
+                        "%Y-%m-%d %H:%M:%S%.f%:z"
+                    };
+                    match UTC.datetime_from_str(&s, fmt) {
+                        Ok(dt) => Ok(dt),
+                        Err(err) => Err(Error::FromSqlConversionFailure(Box::new(err))),
+                    }
+                } else {
+                    NaiveDateTime::column_result(stmt, col).map(|dt| UTC.from_utc_datetime(&dt))
+                }
+            }
+            ffi::SQLITE_INTEGER => {
+                NaiveDateTime::column_result(stmt, col).map(|dt| UTC.from_utc_datetime(&dt))
+            }
+            ffi::SQLITE_FLOAT => {
+                NaiveDateTime::column_result(stmt, col).map(|dt| UTC.from_utc_datetime(&dt))
+            }
+            _ => Err(Error::InvalidColumnType),
+        }
+    }
+
+    unsafe fn column_has_valid_sqlite_type(_: *mut sqlite3_stmt, _: c_int) -> bool {
+        true // to avoid double check
+    }
+}
+
+
+/// ISO 8601 date and time with time zone => "YYYY-MM-DD HH:MM:SS.SSS[+-]HH:MM"
+impl ToSql for DateTime<Local> {
+    unsafe fn bind_parameter(&self, stmt: *mut sqlite3_stmt, col: c_int) -> c_int {
+        let date_str = self.format("%Y-%m-%d %H:%M:%S%.f%:z").to_string();
+        date_str.bind_parameter(stmt, col)
+    }
+}
+
+/// "YYYY-MM-DD HH:MM:SS.SSS[+-]HH:MM"/"YYYY-MM-DD HH:MM"/"YYYY-MM-DD HH:MM:SS"/"YYYY-MM-DD HH:MM:SS.SSS"/ Julian Day / Unix Time => ISO 8601 date and time with time zone.
+/// ("YYYY-MM-DDTHH:MM:SS.SSS[+-]HH:MM"/"YYYY-MM-DDTHH:MM"/"YYYY-MM-DDTHH:MM:SS"/"YYYY-MM-DDTHH:MM:SS.SSS" also supported)
+/// When the timezone is not specified, Local is used.
+impl FromSql for DateTime<Local> {
+    unsafe fn column_result(stmt: *mut sqlite3_stmt, col: c_int) -> Result<DateTime<Local>> {
+        match sqlite3_column_type(stmt, col) {
+            ffi::SQLITE_TEXT => {
+                let s = try!(String::column_result(stmt, col));
+                if s.len() > 23 {
+                    let fmt = if s.as_bytes()[10] == b'T' {
+                        "%Y-%m-%dT%H:%M:%S%.f%:z"
```

Comment:
Is `:z` a typo (should be `%:z`)? Trying to parse `"2016-05-04 17:00:42.862471+08:00"` with `"%Y-%m-%d %H:%M:%S%.f%:z"`, got error `input contains invalid characters`.